sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -414,12 +407,13 @@ object functions {
    * @group agg_funcs
    * @since 1.3.0
    */
-  def count(e: Column): Column = withAggregateFunction {
-    e.expr match {
+  def count(e: Column): Column = {
+    val withoutStar = e.expr match {
       // Turn count(*) into count(1)
```

Comment:
Unrelated issue: this is not the right place to do this conversion. should be done in the analyzer. cc @zhengruifeng does spark connect hit the same issue?

Reply:
No, spark connect doesn't hit such issue.

Reply:
https://github.com/apache/spark/blob/89041a4a8c7b7787fa10f090d4324f20447c4dd3/python/pyspark/sql/connect/functions.py#L1014-L1015

https://github.com/apache/spark/blob/89041a4a8c7b7787fa10f090d4324f20447c4dd3/connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/functions.scala#L391